### PR TITLE
docstring for sundials algs

### PR DESCRIPTION
**Pull request by AnasAbdelR**

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
The full documentation for these methods is pretty long, including all of their arguments and such.

**Comment by AnasAbdelR** [comment]
Do you want me to add everything? There was some code in the doc too

**Comment by AnasAbdelR** [comment]
Can't find the longer documentation for IDA and KINSOL

**Comment by ChrisRackauckas** [comment]
KINSOL doesn't have any 😅 , but IDA is a DAE solver:

https://diffeq.sciml.ai/stable/solvers/dae_solve/#Sundials.jl

File: src/common_interface/algorithms.jl [hunk 1]
```diff
@@ -73,7 +91,24 @@ Base.@pure function CVODE_BDF(;
...
     )
 end
+"""
+CVODE_Adams: CVode Adams-Moulton solver.
+
+method - This is the method for solving the implicit equation. For BDF this defaults to :Newton while for Adams this defaults to :Functional. These choices match the recommended pairing in the Sundials.jl manual. However, note that using the :Newton method may take less iterations but requires more memory than the :Function iteration approach.
+linear_solver - This is the linear solver which is used in the :Newton method.
+The choices for the linear solver are:
 
+:Dense - A dense linear solver.
```
Comment (ChrisRackauckas) [inline src/common_interface/algorithms.jl:101]: any reason why the docs on the other args are cutoff?

Comment (AnasAbdelR) [reply src/common_interface/algorithms.jl:122]: Good catch, the documentation was a little bit non-linear and I missed it xD

**Review by ChrisRackauckas — APPROVED** [review]

**Pull request merged**